zeq/event.h

```diff
@@ -35,7 +38,15 @@ class Event : public boost::noncopyable
     ZEQ_API explicit Event( const uint128_t& type );
 
     /** Move ctor @internal */
+#ifndef BOOST_NO_CXX11_RVALUE_REFERENCES
     ZEQ_API Event( Event&& rhs );
+#else
+    ZEQ_API Event( const Event& rhs )
+    {
+        _impl = rhs._impl;
+        const_cast<Event &>(rhs)._impl = 0;
+    }
```

Comment:
Maybe you're right, but I wonder if all client code can compile without the const modifier, because what you're suggesting is moving the const_cast from the implementation to the client code.